api/opencl_runtime_layer.asciidoc

```diff
@@ -5503,6 +5520,23 @@ Applications must release the memory objects that are acquired using
 commands in the other API.
 This is to guarantee that the state of memory objects is up-to-date and they
 are accessible to the other API.
+
+The following restrictions shall apply -
+  * Each memory object must be released only once. Releasing a memory object 
+    multiple times without acquiring it results in implementation-defined
+    behavior.
```

Comment:
Strictly speaking this is ambiguous (and so is the mirror statement above for acquire operations). There might be a better formulation, possibly describing this in terms of number of acquire/release operations but the current text is a clear step forward so fine if you want to merge it as is.